### PR TITLE
Added DeleteModel functionality and tests

### PR DESCRIPTION
**Pull request by ifielker**
Added DeleteModel functionality and tests

### CODE REVIEW

File: test/integration/machine-learning.spec.ts [hunk 1]
```diff
@@ -31,4 +31,21 @@ describe('admin.machineLearning', () => {
...
+        .should.eventually.be.rejected.and.have.property(
+          'code', 'machine-learning/not-found');
+    });
+
+    it('rejects with invalid-argument when the Model ID is invalid', () => {
+      return admin.machineLearning().deleteModel('invalid-model-id')
+      .should.eventually.be.rejected.and.have.property(
+        'code', 'machine-learning/invalid-argument');
+    });
+
```
Comment (hiranya911) [inline test/integration/machine-learning.spec.ts]: Nit: Remove extra whitespace (empty lines)

File: test/unit/machine-learning/machine-learning-api-client.spec.ts [hunk 2]
```diff
@@ -156,4 +156,82 @@ describe('MachineLearningApiClient', () => {
...
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .rejects(utils.errorFrom(ERROR_RESPONSE, 404));
+      stubs.push(stub);
+      const expected = new FirebaseMachineLearningError('not-found', 'Requested entity not found');
+      return apiClient.deleteModel(MODEL_ID)
+        .should.eventually.be.rejected.and.deep.equal(expected);
+    });
+
+    it('should reject unknown-error when error code is not present', () => {
```
Comment (hiranya911) [inline test/unit/machine-learning/machine-learning-api-client.spec.ts]: Nit: should reject with uknown-error...

File: test/unit/machine-learning/machine-learning-api-client.spec.ts [hunk 3]
```diff
@@ -156,4 +156,82 @@ describe('MachineLearningApiClient', () => {
...
+      const stub = sinon
+        .stub(HttpClient.prototype, 'send')
+        .rejects(utils.errorFrom({}, 404));
+      stubs.push(stub);
+      const expected = new FirebaseMachineLearningError('unknown-error', 'Unknown server error: {}');
+      return apiClient.deleteModel(MODEL_ID)
+        .should.eventually.be.rejected.and.deep.equal(expected);
+    });
+
+    it('should reject unknown-error for non-json response', () => {
```
Comment (hiranya911) [inline test/unit/machine-learning/machine-learning-api-client.spec.ts]: reject with

File: test/unit/machine-learning/machine-learning-api-client.spec.ts [hunk 4]
```diff
@@ -156,4 +156,82 @@ describe('MachineLearningApiClient', () => {
...
+        .stub(HttpClient.prototype, 'send')
+        .rejects(utils.errorFrom('not json', 404));
+      stubs.push(stub);
+      const expected = new FirebaseMachineLearningError(
+        'unknown-error', 'Unexpected response with status: 404 and body: not json');
+      return apiClient.deleteModel(MODEL_ID)
+        .should.eventually.be.rejected.and.deep.equal(expected);
+    });
+
+    it('should reject when rejected with a FirebaseAppError', () => {
```
Comment (hiranya911) [inline test/unit/machine-learning/machine-learning-api-client.spec.ts]: should reject when failed with a network error

File: test/integration/machine-learning.spec.ts (hunk 1 above)
Comment (ifielker) [reply test/integration/machine-learning.spec.ts]: done

File: test/unit/machine-learning/machine-learning-api-client.spec.ts (hunk 2 above)
Comment (ifielker) [reply test/unit/machine-learning/machine-learning-api-client.spec.ts]: done

File: test/unit/machine-learning/machine-learning-api-client.spec.ts (hunk 3 above)
Comment (ifielker) [reply test/unit/machine-learning/machine-learning-api-client.spec.ts]: done

File: test/unit/machine-learning/machine-learning-api-client.spec.ts (hunk 4 above)
Comment (ifielker) [reply test/unit/machine-learning/machine-learning-api-client.spec.ts]: done

**Pull request merged**